src/lib/Libwin/passwd.c

```diff
@@ -1989,8 +2096,10 @@ getlogin(void)
 
 		if (OpenThreadToken(GetCurrentThread(), TOKEN_ALL_ACCESS,
 			TRUE, &token)  == 0) {
-			OpenProcessToken(GetCurrentProcess(), TOKEN_ALL_ACCESS,
-				&token);
+			if (OpenProcessToken(GetCurrentProcess(), TOKEN_ALL_ACCESS,
```

Comment:
since we use getlogin at so many places, could be calling from multiple threads? if so we need to make this function reentrant by removing the static char usern[]

Reply:
I don't recall any place (for Windows) which is multi-threaded. Apart from this if any API fails then as per the code we are setting usern as "".